2019-12-02 22:42:08,121 ERROR: Exception on /explore [GET] [in /Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/flask/app.py:1891]
Traceback (most recent call last):
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/Users/donghao/Desktop/Python/projects/microblog/app/routes.py", line 148, in explore
    return render_template('index.html', title='Explore', posts=posts)
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/flask/templating.py", line 140, in render_template
    ctx.app,
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/flask/templating.py", line 120, in _render
    rv = template.render(context)
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "/Users/donghao/Desktop/Python/projects/microblog/app/templates/index.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/jinja2/environment.py", line 1005, in render
    return concat(self.root_render_func(self.new_context(vars)))
  File "/Users/donghao/Desktop/Python/projects/microblog/app/templates/index.html", line 14, in root
    <p>{{ form.submit() }}</p>
  File "/Users/donghao/Desktop/Python/projects/microblog/app/templates/base.html", line 55, in root
  File "/Users/donghao/Desktop/Python/projects/microblog/app/templates/index.html", line 26, in block_content
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/jinja2/environment.py", line 430, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'form' is undefined
2019-12-02 22:42:14,511 ERROR: Exception on /explore [GET] [in /Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/flask/app.py:1891]
Traceback (most recent call last):
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/Users/donghao/Desktop/Python/projects/microblog/app/routes.py", line 148, in explore
    return render_template('index.html', title='Explore', posts=posts)
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/flask/templating.py", line 140, in render_template
    ctx.app,
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/flask/templating.py", line 120, in _render
    rv = template.render(context)
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "/Users/donghao/Desktop/Python/projects/microblog/app/templates/index.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/jinja2/environment.py", line 1005, in render
    return concat(self.root_render_func(self.new_context(vars)))
  File "/Users/donghao/Desktop/Python/projects/microblog/app/templates/index.html", line 14, in root
    <p>{{ form.submit() }}</p>
  File "/Users/donghao/Desktop/Python/projects/microblog/app/templates/base.html", line 55, in root
  File "/Users/donghao/Desktop/Python/projects/microblog/app/templates/index.html", line 26, in block_content
  File "/Users/donghao/Desktop/Python/projects/microblog/venv/lib/python3.7/site-packages/jinja2/environment.py", line 430, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'form' is undefined
2019-12-02 22:47:20,852 INFO: Microblog startup [in /Users/donghao/Desktop/Python/projects/microblog/app/__init__.py:49]
2019-12-02 22:56:14,931 INFO: Microblog startup [in /Users/donghao/Desktop/Python/projects/microblog/app/__init__.py:49]
2019-12-02 22:57:14,517 INFO: Microblog startup [in /Users/donghao/Desktop/Python/projects/microblog/app/__init__.py:49]
2019-12-02 22:57:41,777 INFO: Microblog startup [in /Users/donghao/Desktop/Python/projects/microblog/app/__init__.py:49]
2019-12-02 22:57:58,354 INFO: Microblog startup [in /Users/donghao/Desktop/Python/projects/microblog/app/__init__.py:49]
2019-12-02 23:02:45,917 INFO: Microblog startup [in /Users/donghao/Desktop/Python/projects/microblog/app/__init__.py:49]
2019-12-02 23:02:59,325 INFO: Microblog startup [in /Users/donghao/Desktop/Python/projects/microblog/app/__init__.py:49]
2019-12-02 23:12:20,008 INFO: Microblog startup [in /Users/donghao/Desktop/Python/projects/microblog/app/__init__.py:52]
2019-12-02 23:17:39,251 INFO: Microblog startup [in /Users/donghao/Desktop/Python/projects/microblog/app/__init__.py:52]
2019-12-02 23:20:26,610 INFO: Microblog startup [in /Users/donghao/Desktop/Python/projects/microblog/app/__init__.py:52]
2019-12-02 23:24:02,163 INFO: Microblog startup [in /Users/donghao/Desktop/Python/projects/microblog/app/__init__.py:52]
2019-12-02 23:44:58,598 INFO: Microblog startup [in /Users/donghao/Desktop/Python/projects/microblog/app/__init__.py:54]
